lib/bugsnag/notification.rb

```diff
@@ -181,7 +181,7 @@ def deliver
         Bugsnag.warn "No API key configured, couldn't notify"
         return
       elsif api_key !~ API_KEY_REGEX
-        Bugsnag.warn "Your API key (#{api_key}) is not valid, couldn't notify"
```

Comment:
It's not dangerous at all to keep it logged, really. The API key cannot be used to access your data in any way. It does seem excessive however that it is logged everywhere always.